File: solutions/julia/grains/1/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
     if square<1 || square> 64
        throw(DomainError(square))
    else
        BigInt(2)^(square -1)
    end
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    if square<1 || square>64
        throw(DomainError(square))
    else
        BigInt(2)^square -1
    end
end

File: solutions/fortran/grains/1/grains.f90

module grains

  implicit none

contains

  double precision function square(n)
    integer :: n
    if ((n>0) .and. (n<65))  then 
      square = 2.d0**(n-1)
    else
      square = -1.d0
    end if
  end function

  double precision function total()
      total = 2.d0**64 - 1
  end function

end module
